fix(integrations): Add `fixes <SHORT-ID>` pitfalls with squashed commits (#5684)
getsentry/sentry-docs: src/docs/product/integrations/source-code-mgmt/bitbucket/index.mdx

Fixes MYAPP-317
```

> When Sentry sees this, we’ll automatically annotate the matching issue with reference to the commit, and later, when that commit is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release, if the commit is squashed Sentry will not know when the commit has been released and the issue may never be marked as a regression.

Reviewer: ```suggestion
When Sentry sees this, we’ll automatically annotate the matching issue with reference to the commit, and later, when that commit is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release. Otherwise, if the commit is squashed, Sentry won’t know when the commit has been released, and the issue may never be marked as a regression.
```
